docs/source/how-to/authentication.md

```diff
@@ -0,0 +1,163 @@
+# Python Client Authentication
+
+This covers authentication from the user (client) perspective. To learn how to
+_deploy_ authenticated Tiled servers, see {doc}`../explanations/security`.
+
+## Interactive Login
+
+Some Tiled servers are configured to let users connect anonymously without
+authenticating.
+
+```py
+>>> from tiled.client import from_uri
+>>> client = from_uri("https://...")
+>>> <Container ...>
+```
+
+Logging in may enable you to see more datasets that may not be public.
+Log in works in one of two ways, depending on the server.
+
+1. Username and password ("OAuth2 password grant")
+
+   ```py
+   >>> client.login()
+   Username: ...
+   Password:
+   ```
+
+2. Via a web browser ("OAuth2 device code grant")
+
+   ```py
+   >>> client.login()
+   You have 15 minutes visit this URL
+
+   https://...
+
+   and enter the code: XXXX-XXXX
+   ```
+
+Some Tiled servers are configured to always require login, disallowing any
+anonymous access. For those, the client will prompt immediately, such as:
+
+```py
+>>> from tiled.client import from_uri
+>>> client = from_uri("https://...")
+Username:
+```
+
+To log out:
+
+```py
+>>> client.logout()
+```
+
+### "Remember Me"
+
+Once you have logged in, if you then close Python and open it later, Tiled will
+log you in automatically, without re-prompting for credentials. It will only
+prompt you again if you log out, or when you session eventually expired.
+
+```py
+>>> from tiled.client import from_uri
+>>> client = from_uri("https://...")
+# Automatically logged in!
+
+# This is a quick way to verify whether you are already logged in
+>>> client.context
+<Context authenticated as '...'>
+```
+
+To opt out of this, set `remember_me=False`:
+
+```py
+>>> from tiled.client import from_uri
+>>> client = from_uri("https://...", remember_me=False)
+```
+
+````{note}
+Tiled stores OAuth2 tokens (it _never_ stores your password) in files
+with properly restricted permissions under `$XDG_CACHE_DIR/tiled/tokens`.
+The path used by your client can be found at:
+
+  ```py
+  >>> client.context.http_client.auth.token_directory
+  ```
+
+To customize the location of this storage, set the environment variable
+`TILED_CACHE_DIR`.
+````
+
+## Noninteractive Authentication (API keys)
+
+There are environments where logging in interactively is not possible,
+such as running a batch script. For these applications, we recommend
+using an API key. These can be created from the CLI:
+
+```sh
+$ tiled login
+$ tiled api_key create
+```
+
+or from an interactive Python session:
+
+```py
+>>> client = from_uri("https://...")
+>>> client.login()
+>>> client.create_api_key()  # {"secret": ...}
+```
+
+The best way to provide an API key is to set the environment variable
+`TILED_API_KEY`. A script like this:
```

Comment:
Would this be a good place to also recommend setting an appropriately "short" lifetime for the token, when it used on shared workstations?

Reply:
Yes, done. While I'm at it, I added support for expressions with units like `"1d"`.